### PR TITLE
Be admin 0200

### PR DESCRIPTION
**Pull request by habibdevsy**
- Captain Financial System entities
- Captain Financial System (get for captain)

### CODE REVIEW

File: backend/src/Controller/Admin/CaptainFinancialSystem/AdminCaptainFinancialSystemAccordingOnOrderController.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+     *      name="token",
+     *      in="header",
+     *      description="token to be passed as a header",
+     *      required=true
+     * )
+     *
+     * @OA\RequestBody(
+     *      description="new Captain's Financial System According On Order",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="integer", property="categoryName"),
```
Comment (ralsalloum) [inline backend/src/Controller/Admin/CaptainFinancialSystem/AdminCaptainFinancialSystemAccordingOnOrderController.php]: categoryName is of type string.

File: backend/src/Controller/Admin/CaptainFinancialSystem/AdminCaptainFinancialSystemAccordingOnOrderController.php [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+     * 
+     * @Security(name="Bearer")
+     */
+    public function createCaptainFinancialSystemAccordingOnOrder(Request $request): JsonResponse
+    {
+        $data = json_decode($request->getContent(), true);
+
+        $request = $this->autoMapping->map(stdClass::class, AdminCaptainFinancialSystemAccordingOnOrderCreateRequest::class, (object)$data);
+
+        $violations = $this->validator->validate($request);
```
Comment (ralsalloum) [inline backend/src/Controller/Admin/CaptainFinancialSystem/AdminCaptainFinancialSystemAccordingOnOrderController.php:92]: FYI: there is no validator for AdminCaptainFinancialSystemAccordingOnOrderCreateRequest in the validation file.

File: backend/src/Request/Admin/CaptainFinancialSystem/AdminCaptainFinancialSystemAccordingToCountOfHoursCreateRequest.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+namespace App\Request\Admin\CaptainFinancialSystem;
+
+class AdminCaptainFinancialSystemAccordingToCountOfHoursCreateRequest
+{
+    private int $countHours;
+    
+    private float $compensationForEveryOrder;
+
+    private float $salary;
+}
```
Comment (ralsalloum) [inline backend/src/Request/Admin/CaptainFinancialSystem/AdminCaptainFinancialSystemAccordingToCountOfHoursCreateRequest.php]: New empty line at the end of the file is needed.

File: backend/src/Controller/Admin/CaptainFinancialSystem/AdminCaptainFinancialSystemAccordingToCountOfHoursController.php [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+     * @OA\Parameter(
+     *      name="token",
+     *      in="header",
+     *      description="token to be passed as a header",
+     *      required=true
+     * )
+     *
+     * @OA\Response(
+     *      response=200,
+     *      description="Returns payments",
```
Comment (ralsalloum) [inline backend/src/Controller/Admin/CaptainFinancialSystem/AdminCaptainFinancialSystemAccordingToCountOfHoursController.php]: description field needs to be updated.

File: backend/src/Controller/Admin/CaptainFinancialSystem/AdminCaptainFinancialSystemAccordingToCountOfOrdersController.php [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+     *          @OA\Property(type="number", property="salary"),
+     *          @OA\Property(type="number", property="monthCompensation"),
+     *          @OA\Property(type="number", property="bounceMaxCountOrdersInMonth"),
+     *          @OA\Property(type="number", property="bounceMinCountOrdersInMonth")
+     *      )
+     * )
+     *
+     * @OA\Response(
+     *      response=201,
+     *      description="Returns Captain's Financial System According On Order",
```
Comment (ralsalloum) [inline backend/src/Controller/Admin/CaptainFinancialSystem/AdminCaptainFinancialSystemAccordingToCountOfOrdersController.php]: description field must be updated to: ... According on count of orders (not orders), right? :)

File: backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingOnOrderController.php [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+use Symfony\Component\Serializer\SerializerInterface;
+use OpenApi\Annotations as OA;
+
+/**
+ * fetch Captain's Financial System According On Order.
+ * @Route("v1/captainfinancialsystemaccordingnorder/")
+ */
+class CaptainFinancialSystemAccordingOnOrderController extends BaseController
+{
+    private CaptainFinancialSystemAccordingOnOrderService $CaptainFinancialSystemAccordingOnOrderService;
```
Comment (ralsalloum) [inline backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingOnOrderController.php]: variable name must start with small letter.

File: backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingOnOrderController.php [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+    public function __construct(SerializerInterface $serializer, CaptainFinancialSystemAccordingOnOrderService $CaptainFinancialSystemAccordingOnOrderService)
+    {
+        parent::__construct($serializer);
+        $this->CaptainFinancialSystemAccordingOnOrderService = $CaptainFinancialSystemAccordingOnOrderService;
```
Comment (ralsalloum) [inline backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingOnOrderController.php]: Please correct the var name here and in the function definition.

File: backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingOnOrderController.php [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+     * @OA\Parameter(
+     *      name="token",
+     *      in="header",
+     *      description="token to be passed as a header",
+     *      required=true
+     * )
+     *
+     * @OA\Response(
+     *      response=200,
+     *      description="Returns payments",
```
Comment (ralsalloum) [inline backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingOnOrderController.php]: description field needs to be changed.

File: backend/src/Service/CaptainFinancialSystem/CaptainFinancialSystemAccordingOnOrderService.php [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+namespace App\Service\CaptainFinancialSystem;
+
+use App\AutoMapping;
+use App\Entity\CaptainFinancialSystemAccordingOnOrderEntity;
+use App\Manager\CaptainFinancialSystem\CaptainFinancialSystemAccordingOnOrderManager;
+use App\Response\CaptainFinancialSystem\CaptainFinancialSystemAccordingOnOrderResponse;
+
+class CaptainFinancialSystemAccordingOnOrderService
+{
+    private CaptainFinancialSystemAccordingOnOrderManager $CaptainFinancialSystemAccordingOnOrderManager;
```
Comment (ralsalloum) [inline backend/src/Service/CaptainFinancialSystem/CaptainFinancialSystemAccordingOnOrderService.php]: Variable name must be small letter. And check all places in which it is being used.

File: backend/src/Response/CaptainFinancialSystem/CaptainFinancialSystemAccordingOnOrderResponse.php [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    public int $countKilometersFrom;
+
+    public int $countKilometersTo;
+
+    public float $amount;
+
+    public float|null $bounce;
+    
+    public float|null $bounceCountOrdersInMonth;
+}
```
Comment (ralsalloum) [inline backend/src/Response/CaptainFinancialSystem/CaptainFinancialSystemAccordingOnOrderResponse.php]: New empty line at the end of the file is needed.

File: backend/src/Manager/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfHoursManager.php [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+use App\Repository\CaptainFinancialSystemAccordingToCountOfHoursEntityRepository;
+use Doctrine\ORM\EntityManagerInterface;
+
+class CaptainFinancialSystemAccordingToCountOfHoursManager
+{
+    private CaptainFinancialSystemAccordingToCountOfHoursEntityRepository $captainFinancialSystemAccordingToCountOfHoursEntityRepository;
+
+    public function __construct(EntityManagerInterface $entityManager, CaptainFinancialSystemAccordingToCountOfHoursEntityRepository $captainFinancialSystemAccordingToCountOfHoursEntityRepository)
+    {
+        $this->entityManager = $entityManager;
```
Comment (ralsalloum) [inline backend/src/Manager/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfHoursManager.php]: entityManager variable is not defined.

File: backend/src/Service/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfHoursService.php [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+namespace App\Service\CaptainFinancialSystem;
+
+use App\AutoMapping;
+use App\Entity\CaptainFinancialSystemAccordingToCountOfHoursEntity;
+use App\Manager\CaptainFinancialSystem\CaptainFinancialSystemAccordingToCountOfHoursManager;
+use App\Response\CaptainFinancialSystem\CaptainFinancialSystemAccordingToCountOfHoursResponse;
+
+class CaptainFinancialSystemAccordingToCountOfHoursService
+{
+    private CaptainFinancialSystemAccordingToCountOfHoursManager $CaptainFinancialSystemAccordingToCountOfHoursManager;
```
Comment (ralsalloum) [inline backend/src/Service/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfHoursService.php]: Please update the variable name, and check all places in which it is being used.

File: backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfHoursController.php [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+use App\Service\CaptainFinancialSystem\CaptainFinancialSystemAccordingToCountOfHoursService;
+use Nelmio\ApiDocBundle\Annotation\Security;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\SerializerInterface;
+use OpenApi\Annotations as OA;
+
+/**
+ * Create and fetch Captain's Financial System According To Count Of Hours.
```
Comment (ralsalloum) [inline backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfHoursController.php]: route description needs to be updated (there is no create APIs here).

File: backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfHoursController.php [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+ * @Route("v1/captainfinancialsystemaccordingtocountofhours/")
+ */
+class CaptainFinancialSystemAccordingToCountOfHoursController extends BaseController
+{
+    private CaptainFinancialSystemAccordingToCountOfHoursService $CaptainFinancialSystemAccordingToCountOfHoursService;
```
Comment (ralsalloum) [inline backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfHoursController.php]: Change the variable name (can not start with capital letter), and check all places in which it is being used.

File: backend/src/Service/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfOrdersService.php [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+namespace App\Service\CaptainFinancialSystem;
+
+use App\AutoMapping;
+use App\Entity\CaptainFinancialSystemAccordingToCountOfOrdersEntity;
+use App\Manager\CaptainFinancialSystem\CaptainFinancialSystemAccordingToCountOfOrdersManager;
+use App\Response\CaptainFinancialSystem\CaptainFinancialSystemAccordingToCountOfOrdersResponse;
+
+class CaptainFinancialSystemAccordingToCountOfOrdersService
+{
+    private CaptainFinancialSystemAccordingToCountOfOrdersManager $CaptainFinancialSystemAccordingToCountOfOrdersManager;
```
Comment (ralsalloum) [inline backend/src/Service/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfOrdersService.php]: Alter the first letter of the var name from capital to small letter, and check all places in which it is being used.

File: backend/src/Response/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfOrdersResponse.php [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+    public int $countOrdersInMonth;
+
+    public float $salary;
+
+    public float $monthCompensation;
+
+    public float $bounceMaxCountOrdersInMonth;
+    
+    public float $bounceMinCountOrdersInMonth;
+}
```
Comment (ralsalloum) [inline backend/src/Response/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfOrdersResponse.php]: Add new empty line at the end of the file please.

File: backend/src/Manager/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfOrdersManager.php [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+use Doctrine\ORM\EntityManagerInterface;
+use App\Repository\CaptainFinancialSystemAccordingToCountOfOrdersEntityRepository;
+
+class CaptainFinancialSystemAccordingToCountOfOrdersManager
+{
+    private CaptainFinancialSystemAccordingToCountOfOrdersEntityRepository $captainFinancialSystemAccordingToCountOfOrdersEntityRepository;
+
+    public function __construct(EntityManagerInterface $entityManager, CaptainFinancialSystemAccordingToCountOfOrdersEntityRepository $captainFinancialSystemAccordingToCountOfOrdersEntityRepository)
+    {
+        $this->entityManager = $entityManager;
```
Comment (ralsalloum) [inline backend/src/Manager/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfOrdersManager.php]: entityManager is being used but not defined.

File: backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfOrdersController.php [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+use Symfony\Component\Serializer\SerializerInterface;
+use OpenApi\Annotations as OA;
+
+/**
+ * Create and fetch Captain's Financial System According To Count Of Orders.
+ * @Route("v1/captainfinancialsystemaccordingntocountoforders/")
+ */
+class CaptainFinancialSystemAccordingToCountOfOrdersController extends BaseController
+{
+    private CaptainFinancialSystemAccordingToCountOfOrdersService $CaptainFinancialSystemAccordingToCountOfOrdersService;
```
Comment (ralsalloum) [inline backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfOrdersController.php]: update the var name (can not start with capital letter), and change it also in all places in which it is being used.

File: backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfOrdersController.php [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+use Nelmio\ApiDocBundle\Annotation\Security;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\SerializerInterface;
+use OpenApi\Annotations as OA;
+
+/**
+ * Create and fetch Captain's Financial System According To Count Of Orders.
+ * @Route("v1/captainfinancialsystemaccordingntocountoforders/")
```
Comment (ralsalloum) [inline backend/src/Controller/CaptainFinancialSystem/CaptainFinancialSystemAccordingToCountOfOrdersController.php:16]: mmm there is no create APIs here. Please update the route description.

**Review by ralsalloum — DISMISSED** [review]
Please resolve the requested changes.

**Review by Kenan-Hussein — APPROVED** [review]

**Pull request merged**